packages/runtime/src/runtime/hooks/useEnsembleAction.tsx

```diff
@@ -299,6 +315,179 @@ export const useInvokeAPI: EnsembleActionHook<InvokeAPIAction> = (action) => {
   return invokeApi;
 };
 
+export const useConnectSocket: EnsembleActionHook<ConnectSocketAction> = (
+  action,
+) => {
+  const { sockets, setData } = useScreenData();
+  const [isComplete, setIsComplete] = useState<boolean>();
+  const [isLoading, setIsLoading] = useState<boolean>();
```

Comment:
Change isComplete to isOpen. Socket lifecycle is different from normal HTTP request